### PR TITLE
Adds Why Unity

### PR DESCRIPTION
**Pull request by caseyyee**
Fixes https://github.com/mozilla/unity-webvr-export/issues/149

### CODE REVIEW

File: docs/explainer.md [hunk 1]
```diff
@@ -1,8 +1,38 @@
-# Troubleshooting & FAQ
+# WebVR Unity Assets Explained
 
 ## What is WebVR?
 
-The WebVR assets use the WebVR browser API's, a standard-based implementation that removes the need for any platform specific SDK’s and also provides ability to be responsive to different VR configurations.
+The WebVR asset package uses the WebVR browser API's, a standard-based implementation that removes the need for any platform specific SDK’s and also provides ability to be responsive to different VR configurations.
```
Comment (cvan) [inline docs/explainer.md]: - `The WebVR asset package` -> `The WebVR Assets package`
- `API's` -> `APIs`
- `a standard-based` -> `a standards-based`
- `platform specific` -> `platform-specific`
- `SDK’s` -> `SDKs`
- `and also provides ability to be responsive to different VR configurations` -> `and handles responsiveness for a variety of VR configurations` (or similar)

File: docs/explainer.md [hunk 2]
```diff
@@ -1,8 +1,38 @@
...
+
+[See WebVR specification](https://immersive-web.github.io/webvr/spec/1.1/)
```
Comment (cvan) [inline docs/explainer.md]: can you bold this and add a period?

File: docs/explainer.md [hunk 3]
```diff
@@ -1,8 +1,38 @@
...
+
+## What browsers and headsets are supported?
+
+[See brower and headset compatibility](https://github.com/mozilla/unity-webvr-export#unity-compatibility)
```
[the remark this answers is not in the document]
Comment (cvan) [reply docs/explainer.md]: can you bold this and add a period?

File: docs/explainer.md [hunk 4]
```diff
@@ -1,8 +1,38 @@
...
+
+## Why Unity and not [A-Frame](https://aframe.io/), [three.js](https://threejs.org/) or some other framework?
+
+Each has it's own set of advantages, but all share the same [benefits of publishing to the web](https://github.com/mozilla/unity-webvr-export/blob/master/docs/troubleshooting-faq.md#why-wouldnt-i-just-create-a-native-app) and implement the [WebVR API's](https://immersive-web.github.io/webvr/spec/1.1/).
```
[the remark this answers is not in the document]
Comment (cvan) [reply docs/explainer.md]: `API's` -> `APIs`

File: docs/explainer.md [hunk 5]
```diff
@@ -1,8 +1,38 @@
...
+
+This is a _very_ basic overview of maybe some of the reasons why you would choose one option over another, we encourage creators to evaluate all the options to make a fully informed decision based on your specific needs:
```
[the remark this answers is not in the document]
Comment (cvan) [reply docs/explainer.md]: feel free to remove `of maybe `

File: docs/explainer.md [hunk 6]
```diff
@@ -1,8 +1,38 @@
...
+
+**Unity**
+
+_Pro:_ You are familiar with working with Unity and want to take advantage of its graphics and physics capabilities, large community of 3rd party assets and resources, or have existing content that you would like to publish to the web.
+
+_Con:_ Slower to load, closed source platform, lacks robust mobile support.
```
Comment (cvan) [inline docs/explainer.md]: `closed source platform` -> `closed-source engine`

File: docs/explainer.md [hunk 7]
```diff
@@ -1,8 +1,38 @@
...
+
+## Why Unity and not [A-Frame](https://aframe.io/), [three.js](https://threejs.org/) or some other framework?
+
+Each has it's own set of advantages, but all share the same [benefits of publishing to the web](https://github.com/mozilla/unity-webvr-export/blob/master/docs/troubleshooting-faq.md#why-wouldnt-i-just-create-a-native-app) and implement the [WebVR API's](https://immersive-web.github.io/webvr/spec/1.1/).
+
+This is a _very_ basic overview of maybe some of the reasons why you would choose one option over another, we encourage creators to evaluate all the options to make a fully informed decision based on your specific needs:
+
+**Unity**
+
+_Pro:_ You are familiar with working with Unity and want to take advantage of its graphics and physics capabilities, large community of 3rd party assets and resources, or have existing content that you would like to publish to the web.
```
Comment (cvan) [inline docs/explainer.md]: `3rd party assets` -> `third-party assets`

File: docs/explainer.md [hunk 8]
```diff
@@ -1,8 +1,38 @@
...
+
+[See brower and headset compatibility](https://github.com/mozilla/unity-webvr-export#unity-compatibility)
+
+## Why Unity and not [A-Frame](https://aframe.io/), [three.js](https://threejs.org/) or some other framework?
+
+Each has it's own set of advantages, but all share the same [benefits of publishing to the web](https://github.com/mozilla/unity-webvr-export/blob/master/docs/troubleshooting-faq.md#why-wouldnt-i-just-create-a-native-app) and implement the [WebVR API's](https://immersive-web.github.io/webvr/spec/1.1/).
+
+This is a _very_ basic overview of maybe some of the reasons why you would choose one option over another, we encourage creators to evaluate all the options to make a fully informed decision based on your specific needs:
+
+**Unity**
```
Comment (cvan) [inline docs/explainer.md]: perhaps this isn't the place for it, but perhaps a brief FAQ could be `Is there a Unreal Engine WebVR Exporter?`

File: docs/explainer.md [hunk 9]
```diff
@@ -1,8 +1,38 @@
...
+
+_Pro:_ You are familiar with working with Unity and want to take advantage of its graphics and physics capabilities, large community of 3rd party assets and resources, or have existing content that you would like to publish to the web.
+
+_Con:_ Slower to load, closed source platform, lacks robust mobile support.
+
+**A-Frame or three.js**
+
+_Pro:_ You're a Javascript developer and familiar with web development workflows, fast to load, open source and good mobile support.
```
Comment (cvan) [inline docs/explainer.md]: - `Javascript` -> `JavaScript`
- `web development workflows` -> `web-development workflows`

File: README.md [hunk 10]
```diff
@@ -39,7 +39,7 @@ This asset works by utilizing Unity's WebGL platform support and therefore share
 
 * [Setting up a Unity project for WebVR](./docs/project-setup.md)
 * [Publishing](./docs/publishing.md)
-* [Troubleshooting and FAQ](./docs/troubleshooting-faq.md)
+* [WebVR Unity Assets Explained](./docs/explainer.md)
```
[the remark this answers is not in the document]
Comment (caseyyee) [reply README.md]: Haha I don't know.   that's how we've been describing it.  

Maybe "Asset Package"?  to be more precise?

[the remark this answers is not in the document]
Comment (cvan) [reply README.md]: `Explanation of Unity WebVR Assets package`

File: docs/explainer.md (hunk 8 above)
Comment (caseyyee) [reply docs/explainer.md]: Hmm. yeah, will think about where we might be able to fit that in.

**Comment by caseyyee** [comment]
@cvan good for another r?

File: README.md [hunk 11]
```diff
@@ -1,4 +1,4 @@
-# Unity WebVR Assets
+# Explanation of Unity WebVR Assets package
```
Comment (delapuente) [inline README.md]: It is confusing that the README is titled `Explanation of Unity WebVR Assets package` and then link to another document using this title as the anchor name and a slightly different title for the new document. My recommendation is to keep the original title for the README.

File: README.md [hunk 12]
```diff
@@ -39,7 +39,7 @@ This asset works by utilizing Unity's WebGL platform support and therefore share
 
 * [Setting up a Unity project for WebVR](./docs/project-setup.md)
 * [Publishing](./docs/publishing.md)
-* [Troubleshooting and FAQ](./docs/troubleshooting-faq.md)
+* [Explanation of Unity WebVR Assets package](./docs/explainer.md)
```
Comment (delapuente) [inline README.md]: I highly recommend to undo this change. `Troubleshooring and FAQ` is a great name for reference. If the user is in trouble, she will probably look for it. It is less obvious to look for the _explainer_ when you need help.

File: docs/explainer.md [hunk 13]
```diff
@@ -1,8 +1,39 @@
-# Troubleshooting & FAQ
+# WebVR Unity Assets Explained
```
Comment (delapuente) [inline docs/explainer.md]: I would leave "Trobuleshooting & FAQ" for the reasons explained above.

File: docs/explainer.md [hunk 14]
```diff
@@ -1,8 +1,39 @@
...
-The WebVR assets use the WebVR browser API's, a standard-based implementation that removes the need for any platform specific SDK’s and also provides ability to be responsive to different VR configurations.
+The WebVR Assets package uses the WebVR browser APIs, a standards-based implementation that removes the need for any platform-specific SDKs and handles responsiveness for a variety of VR configurations.
+
+**[See WebVR specification.](https://immersive-web.github.io/webvr/spec/1.1/)**
+
+## What browsers and headsets are supported?
+
+**[See browser and headset compatibility.](https://github.com/mozilla/unity-webvr-export#unity-compatibility)**
+
+## Why Unity and not [A-Frame](https://aframe.io/), [three.js](https://threejs.org/) or some other framework?
```
Comment (delapuente) [inline docs/explainer.md]: I highly recommend focussing on `Why Unity` and **nothing else**. Explain the same you explain in the [Hacks post](https://hacks.mozilla.org/2018/02/create-vr-on-the-web-using-unity3d/):
> With the release of these assets, we hope to bridge the frictionless distribution, ease of use and accessibility of the Web with the best-in-class content creation tools from Unity.

If you are not comparing A-Frame, Three.js, Babylon... in-depth, don't do it at all. Mentioning one _pro_ and one _con_ is incomplete and seems biased. Furthermore, Unity and Web libraries don't target the same audience.

File: docs/explainer.md [hunk 15]
```diff
@@ -1,8 +1,39 @@
...
+
+_Con:_ Slower to load, closed-source engine, lacks robust mobile support.
+
+**A-Frame or three.js**
+
+_Pro:_ You're a JavaScript developer and familiar with web-development workflows, fast to load, open source and good mobile support.
+
+_Con:_ Still early in its evolution, lacks some of the major features of a developed game engine like Unity.
+
+**Some other great ways to make WebVR content**
```
Comment (delapuente) [inline docs/explainer.md]: I would add `Three.js` and `A-Frame` here and remove the poor comparison we do before.

File: README.md (hunk 12 above)
Comment (caseyyee) [reply README.md]: OK, agree with this.  I'll change it back.  Clearer in the broader context of the README.md
Thanks

File: docs/explainer.md (hunk 14 above)
Comment (caseyyee) [reply docs/explainer.md]: It's a common question that has come up, so I think it's worthwhile to have _something_.   How do you think we should characterize the difference?

Comment (cvan) [reply docs/explainer.md]: since I've seen the question come up on Reddit and Twitter more than a few times, I think it's worth mentioning in the docs. it doesn't have to be front and center on the README (as it's not here).

how about replacing the section with something like this?

(I spent some time on this, but feel free to chop it up all you want!)

https://gist.github.com/cvan/ad492b46d2624a4e3cd2d9f6d7878bf6#readme

```
## Why Unity?

Unity is the most popular game engine in the indie game-developer community. There is a huge community out there to support creators, in particular a massive collection of assets in the [Asset Store](https://assetstore.unity.com/), which reduces the time and skills needed to build a game. Unity has a relatively intuitive interface and cross-platform integration (including WebGL!) so developers can target multiple platforms using a single project codebase.

[Unity's Personal Edition](https://store.unity.com/products/unity-personal) is completely free to use. The Unity splash screen is required to be shown, and you cannot use realtime developer analytics nor Unity Cloud Build services.

Disclaimer about mobile support: While Unity is the most popular native game engine on mobile, Unity's WebGL player is particularly large and experimental to use because of the sheer limited processing capabilities of today's mobile devices.

## Other Engines

Epic Games' [Unreal Engine](https://en.wikipedia.org/wiki/Unreal_Engine) has a steeper learning curve, but it offers better graphical capabilities, has an open-source engine and is better suited for AAA-quality games. If you are targeting Unreal Engine, there is a [built-in HTML5 packager](https://docs.unrealengine.com/latest/INT/Platforms/HTML5/GettingStarted/index.html), which uses [emscripten](https://kripken.github.io/emscripten-site/) to convert the engine C++ to JavaScript (WebGL). WebVR is not supported.

If you're a web developer, there are several popular JavaScript tools and frameworks you can use to create your VR experience for the Web:

* [A-Frame](https://aframe.io/)
* [three.js](https://threejs.org/)
* [ReactVR](https://facebook.github.io/react-vr/)
* [Babylon.js](https://www.babylonjs.com/)
* [PlayCanvas](https://playcanvas.com/)
* [JanusVR](http://janusvr.com/)
* [Amazon Sumerian](https://aws.amazon.com/sumerian/)
```

File: docs/explainer.md (hunk 15 above)
Comment (cvan) [reply docs/explainer.md]: see my comment above

**Comment by delapuente** [comment]
@cvan I really like your answer. It reminds me of your Hacks post review and the reasons you enumerated there are the same we should speak of here. I would remove mentioning Unreal Engine, though.

**Comment by caseyyee** [comment]
@cvan @delapuente great suggestions, will make updates.  Thanks!

**Comment by caseyyee** [comment]
Thanks @cvan @delapuente for helping refine this.

File: docs/troubleshooting-faq.md [hunk 16]
```diff
@@ -14,9 +40,9 @@ The web makes it easy to deliver content to users. Just navigate or link to a ex
 
 ## Will it work on mobile?
 
-The package targets desktop VR browsers.   Mobile device support is limited and not recommended. Please see [compatibility section](../README.md) for more information.
+The WebVR Assets package targets desktop VR browsers. Mobile VR browser support is limited and not recommended at this time due to the performance and memory limitations of the Unity WebGL player on mobile browsers. Although we have had some luck running on Android devices, Safari on IOS fails to load.  We are investigate ways to make the exporter more useable on mobile.
```
Comment (cvan) [inline docs/troubleshooting-faq.md]: nit: `IOS` -> `iOS`

Comment (cvan) [inline docs/troubleshooting-faq.md]: typos:

- `IOS` -> `iOS`
- `investigate` -> `investigating`
- `useable` -> `usable`

Comment (cvan) [reply docs/troubleshooting-faq.md]: I just made those edits in bdf74b9

**Comment by cvan** [comment]
good work, @caseyyee!
